editoast/src/modelsv2/rolling_stock_model.rs

```diff
@@ -0,0 +1,128 @@
+use actix_web::web::Data;
+use derivative::Derivative;
+use diesel::result::Error as DieselError;
+use diesel::{ExpressionMethods, QueryDsl, SelectableHelper};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use diesel_json::Json as DieselJson;
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use utoipa::ToSchema;
+use validator::{Validate, ValidationError};
+
+use crate::error::Result;
+use crate::models::rolling_stock::rolling_stock_livery::RollingStockLiveryMetadata;
+use crate::schema::rolling_stock::{
+    EffortCurves, EnergySource, Gamma, RollingResistance, RollingStockMetadata,
+    RollingStockWithLiveries,
+};
+use crate::DbPool;
+
+#[derive(Clone, Debug, Deserialize, Serialize, ModelV2, Derivative, ToSchema, Validate)]
+#[derivative(Default, PartialEq)]
+#[model(table = crate::tables::rolling_stock)]
+#[validate(schema(function = "validate_rolling_stock"))]
+pub struct RollingStockModel {
+    pub id: i64,
+    pub railjson_version: String,
+    #[derivative(PartialEq = "ignore")]
+    #[schema(value_type = String)]
+    pub name: String,
+    #[schema(value_type = EffortCurves)]
+    pub effort_curves: DieselJson<EffortCurves>,
+    #[derivative(PartialEq = "ignore")]
+    #[schema(value_type = RollingStockMetadata)]
+    pub metadata: DieselJson<RollingStockMetadata>,
+    pub length: f64,
+    pub max_speed: f64,
+    pub startup_time: f64,
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    #[schema(value_type = Gamma)]
+    pub gamma: DieselJson<Gamma>,
+    pub inertia_coefficient: f64,
+    pub base_power_class: Option<String>,
+    #[schema(value_type = Vec<String>)]
+    pub features: Vec<Option<String>>, // Previously it was TextArray
+    pub mass: f64,
+    #[schema(value_type = RollingResistance)]
+    pub rolling_resistance: DieselJson<RollingResistance>,
+    #[schema(value_type = LoadingGaugeType)]
+    pub loading_gauge: String,
+    #[schema(value_type = Option<HashMap<String, String>>)]
+    pub power_restrictions: Option<DieselJson<HashMap<String, String>>>,
```

Comment:
```suggestion
    #[model(json)]
    pub power_restrictions: Option<HashMap<String, String>>,
```

This is not strictly equivalent (`NULL != 'null'::jsonb`) so there's probably some changes to make at insertion, and a migration to always have a JSON value.